tools/codegen/gen.py

```diff
@@ -1007,69 +996,60 @@ def make_file_manager(install_dir: str) -> FileManager:
 #include <ATen/hip/HIPDevice.h>
 #include <ATen/hip/HIPContext.h>'''
 
-    backends = [
+    # NB: substrings in these dispatch keys matter, we do tests to see if
+    # a key contains, e.g., CUDA to classify it as a CUDA backend
+    dispatch_keys = [
```

Comment:
Btw a quick comment since I ran into this PR ;) 
I tried to use dispatch_keywords to reference the keys we support in native_functions.yaml to disambiguate from the real dispatch keys. One main confusion if we keep using dispatch_keys is developers would naturally think all dispatch keys are supported (like XLA/Autograd/Autocast etc) but in fact not all dispatch keys are supported in native_functions.yaml.

Reply:
Ah, so it wasn't just a typo!

Here's a take: what if we hypothetically, we should support all dispatch keys in `native_functions.yaml`? Then dispatch key naming would be appropriate.